joomla/joomla-cms
# Extend the color picker to change between hex, rgb, rgba

Pull Request for Issue #14353 .

### Summary of Changes

Adds buttons to change between hex, rgb, rgba.

### Testing Instructions

Add to protostar templateDetails.xml code below to the xml `extension -> config -> fields` and go to settings of that template and then to the advanced tab. There should be new extended color pickers.

```xml
<field name="Color1" type="color" label="Color + changeformat" changeformat="true" />
<field name="Color2" type="color" label="Color + keywords" format="hex" keywords="transparent,inherit,initial" />
<field name="Color3" type="color" label="Color + format" format="rgb" />
<field name="Color4" type="color" label="Color" />
```
`keywords` - if you add this attribute to field then you can type to the input keywords that you specify and save it.
`format` - you can choose color only from specified format of color.
`changeformat` - it enables you to save color in 3 formats `rgb`, `hex` and `rgba`

### Documentation Changes Required

Yes on the page [https://docs.joomla.org/Color_form_field_type](https://docs.joomla.org/Color_form_field_type)

- boolean attribute `changeformat=""` is to show buttons to change format between hex, rgb, rgba.
- format and keywords attribute from #10129

![zaznaczenie_134](https://cloud.githubusercontent.com/assets/8504663/23776318/e8e6e66a-052d-11e7-855c-2741740f6ac1.png)
